getstatus.js

```diff
@@ -0,0 +1,135 @@
+"use strict";
+// @ts-check
+Object.defineProperty(exports, "__esModule", { value: true });
+
+const api_1 = require("@polkadot/api");
+const types_1 = require("@joystream/types");
+const BN = require('bn.js');
+
+async function getStatusUpdate () {
+  let update = {}
+
+   // Initialise the provider to connect to the local node
+   //const provider = new api_1.WsProvider('wss://staging-reckless.joystream.org/reckless/rpc/');
+   const provider = new api_1.WsProvider('wss://staging-lts.joystream.org/staging/rpc/');
+   //const provider = new api_1.WsProvider('ws://127.0.0.1:9944');
+   // register types before creating the api
+   types_1.registerJoystreamTypes();
+   // Create the API and wait until ready
+   const api = await api_1.ApiPromise.create(provider);
+   
+   
+   // Retrieve the chain & node information information via rpc calls
+   const [chain, nodeName, nodeVersion, peers] = await Promise.all([
+       api.rpc.system.chain(),
+       api.rpc.system.name(),
+       api.rpc.system.version(),
+       api.rpc.system.peers()
+    ]);
+
+    update.system = {
+      'chain': chain,
+      'name': nodeName,
+      'version': nodeVersion,
+      //'peerCount': peerCount.length
+      'peerCount': peers.length
+    }
+
+    // Retrieve finalized block height
+    const finalizedHash = await api.rpc.chain.getFinalizedHead();
+    const finalizedBlockNumber = await api.rpc.chain.getHeader(`${finalizedHash}`);
+    update.block_height = (finalizedBlockNumber.blockNumber)
+
+    // Retrieve runtime data
+    const [spec, impl, specVer] = await Promise.all([
+      api.rpc.chain.getRuntimeVersion(`${finalizedHash}`),
+      api.rpc.chain.getRuntimeVersion(`${finalizedHash}`),
+      api.rpc.chain.getRuntimeVersion(`${finalizedHash}`)
+   ]);
+
+   update.runtime_version = {
+    'spec_name': spec.specName,
+    'impl_name': impl.implName,
+    'spec_version': specVer.specVersion
+  }
+  
+  // Retrieve council and election data
+  const [councilMembers, electionStage] = await Promise.all([
+    api.query.council.activeCouncil(),
+    api.query.councilElection.stage()
+  ]);
+  var stage = (`${electionStage}`);
```

Comment:
That was a clever hack I have to admit, but you probably didn't realize the helper properties available on the election stage type, as you were not working in typescript environment.

I can suggest the following simplification:

```js
update.council = {
    'members_count': councilMembers.length,
    'election_stage': electionStage.isSome ? electionStage.unwrap().type : "Not Running"
  }
```

Reply:
That does look _somewhat_ better :P 
I tried something very similar first, but withouth the `.unwrap()`, so if failed!